packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -110,3 +110,56 @@ export async function storeEvents(
   await dynamoDB.batchWrite(params).promise()
   logger.debug('[EventsAdapter#storeEvents] EventEnvelope stored')
 }
+
+export async function destroyEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entityTypeName: string,
+  entityID: UUID
+): Promise<void> {
+  logger.debug('[EventsAdapter#destroyEntity] Destroying entity')
+
+  // We need to fetch the entity events and snapshots
+  const events = await readEntityEventsSince(dynamoDB, config, logger, entityTypeName, entityID)
+  const snapshots = await readEntityEventsSince(
+    dynamoDB,
+    config,
+    logger,
+    entityTypeName,
+    entityID,
+    originOfTime,
+    'snapshot'
+  )
```

Comment:
I think that you could get both in a single query and avoid modifying the `readEntityEventsSince` function by looking for all the objects in the events table that share the same entityTypeName and entityID:

```typescript
const eventsAndSnapshots = await dynamoDB
     .query({
       TableName: config.resourceNames.eventsStore,
       ConsistentRead: true,
       KeyConditionExpression: `${eventsStoreAttributes.partitionKey} = :partitionKey`,
       ExpressionAttributeValues: {
         ':partitionKey': partitionKeyForEvent(entityTypeName, entityID),
       },
       ScanIndexForward: true, // Ascending order (older timestamps first)
    })
    .promise()
```

Also, I don't think you need to set the `ScanindexForward` parameter, it doesn't really matter in which order you delete them.

Reply:
Regarding `ScanIndexForward` since I'm using an existing method, I need to check the implications of not setting it. Let me check.

Reply:
@javiertoledo The problem with our partition key is that contains also de `kind` field. So we can get the snapshots and events at the same time. Another option could be a `scan` instead of a `query`. But we should take in into account the cost of this operation, anyway who is deleting things all the time? Thoughts?

Reply:
What if we don't rely on querying first, just delete the whole thing?

Reply:
Gotcha, in that case, I think it's better to keep your current implementation. This use case should be avoided in most situations, you should have a very good reason to delete past events, like complying with GDPR regulations or something like that, so it's okay-ish if it's slow. What we have to make sure of is that it doesn't fail. I think we should at least create an issue to review this in the future to implement pagination or use step functions or something like that to make sure that the process doesn't fail for long event streams.